### PR TITLE
Add global position msl option.

### PR DESCRIPTION
**Pull request by jgoppert**
For a project I needed global position in msl so added this.

### CODE REVIEW

**Review by TSC21 — COMMENTED** [review]
This goes against the convention used in ROS. Should this be exposed in a different message?

File: mavros/src/plugins/global_position.cpp [hunk 1]
```diff
@@ -144,7 +147,10 @@ class GlobalPositionPlugin : public plugin::PluginBase {
 	{
 		fix->latitude = msg.lat / 1E7;		// deg
 		fix->longitude = msg.lon / 1E7;		// deg
-		fix->altitude = msg.alt / 1E3 + m_uas->geoid_to_ellipsoid_height(fix);	// in meters
+		fix->altitude = msg.alt / 1E3;      // in meters
+		if (!use_msl_alt) {
+			m_uas->geoid_to_ellipsoid_height(fix);	// offset for ellipsoid
```
Comment (Jaeyoung-Lim) [inline mavros/src/plugins/global_position.cpp:152]: Shouldn't this be added to `altitude`?

**Comment by jgoppert** [comment]
@TSC21 I was looking at this REP, https://www.ros.org/reps/rep-0105.html, can you direct me to the one you are referencing?
@Jaeyoung-Lim  Thanks for the catch.

**Comment by Jaeyoung-Lim** [comment]
@TSC21 @jgoppert Any updates?

**Comment by Jaeyoung-Lim** [comment]
@vooon This PR seems stale, any reason to keep it open?

**Comment by vooon** [comment]
I suppose that should be made different. IMHO better to publish MSL in different topic, like Float64Stamped and use message_filters where it is needed.

According to [NavSatFix](https://docs.ros.org/en/api/sensor_msgs/html/msg/NavSatFix.html) altitude *must* be in WGS84.
So MSL he might be surprising for other people.

**Pull request closed without merging**